### PR TITLE
Update claims filter text as filters get applied.

### PR DESCRIPTION
**Pull request by hinzed1127**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**: Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
- _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Updates the paginated results description as filters and sorting are changed/applied

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/87886

## Testing done

- Added unit tests to assert proper UI appearance, as well as fixed/updated existing tests 

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

**Single Filter**

Screenshot 2024-07-30 at 5 25 08 PM

**Multiple Filters**

Screenshot 2024-07-30 at 5 25 54 PM

**Zero claims with chosen filters**

Screenshot 2024-07-30 at 5 23 25 PM

## What areas of the site does it impact?

Travel Pay

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by rmessina1010 — APPROVED** [review]
LGTM

File: src/applications/travel-pay/containers/TravelPayStatusApp.jsx [hunk 1]
```diff
@@ -180,6 +180,41 @@ export default function App({ children }) {
...
+
+    if (appliedFiltersLength > 0) {
+      sortAndFilterText += `, with ${appliedFiltersLength} ${
+        appliedFiltersLength === 1 ? 'filter' : 'filters'
+      } applied`;
+    }
+
+    if (numResults === 0 && appliedFiltersLength > 0) {
+      // Note that appliedFiltersLength === 1 should never be true here, since
+      // 0 claims matching a single filter should mean the filter isn't shown
```
Comment (athifw) [inline src/applications/travel-pay/containers/TravelPayStatusApp.jsx:207]: `appliedFiltersLength == 1` is possible if one of the hardcoded status filters was applied, and it's possible that none of the claims match the hardcoded status

**Review by athifw — APPROVED** [review]
Approved, with one comment

File: src/applications/travel-pay/containers/TravelPayStatusApp.jsx (hunk 1 above)
Comment (hinzed1127) [reply src/applications/travel-pay/containers/TravelPayStatusApp.jsx:207]: If I understand what you're referring to, hardcoded filters got removed in https://github.com/department-of-veterans-affairs/vets-website/pull/30877 and made `vets-api`/the upstream travel pay API the source of truth of status filter names, which I think means this comment is accurate? I could be missing something, though.

Comment (athifw) [reply src/applications/travel-pay/containers/TravelPayStatusApp.jsx:207]: By hardcoded filters, I meant the statuses we always show in the filters no matter what. These statuses are: `'On Hold', 'Denied', 'In Manual Review'`
So for example, it's possible for none of the claims to be `In Manual Review`, but the `In Manual Review` filter would still be displayed.

Comment (hinzed1127) [reply src/applications/travel-pay/containers/TravelPayStatusApp.jsx:207]: I see what you're saying now. While we _are_ hardcoding the top statuses, [we're still filtering them out](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/travel-pay/containers/TravelPayStatusApp.jsx#L56-L62) if they're not present in any of the returned claims

**Pull request merged**